### PR TITLE
Ban offline users

### PR DESCRIPTION
**Pull request by 01-Feli**
Added support for banning offline users

### CODE REVIEW

**Comment by 01-Feli** [comment]
This will close https://github.com/openmod/openmod/issues/461

File: unturned/OpenMod.Unturned/Users/UnturnedUserProvider.cs [hunk 1]
```diff
@@ -208,6 +211,15 @@ protected virtual void OnPendingPlayerConnecting(ValidateAuthTicketResponse_t ca
 
                 userConnectingEvent.IsCancelled = !isPendingValid;
 
+                if (userData != null && userData.UnBan != null && userData.UnBan >= DateTime.Now)
+                    await userConnectingEvent.RejectAsync(Smart.Format("You are banned from this server until the {Day} of {Month} at {Hour}:{Minute}", new
```
Comment (Trojaner) [inline unturned/OpenMod.Unturned/Users/UnturnedUserProvider.cs:215]: Must be translatable

File: framework/OpenMod.API/Users/UserData.cs [hunk 2]
```diff
@@ -34,6 +34,11 @@ public sealed class UserData
         /// </summary>
         public DateTime? LastSeen { get; set; }
 
+        /// <summary>
+        /// Gets or sets the unban date, this is used for banning offline players
+        /// </summary>
+        public DateTime? UnBan { get; set; }
```
Comment (Trojaner) [inline framework/OpenMod.API/Users/UserData.cs:40]: Rename to UnbanTime instead

**Review by Trojaner — CHANGES_REQUESTED** [review]

File: unturned/OpenMod.Unturned/Users/UnturnedUserProvider.cs (hunk 1 above)
Comment (rube200) [reply unturned/OpenMod.Unturned/Users/UnturnedUserProvider.cs:215]: Why dont you pass the reason??

File: framework/OpenMod.API/Users/UserData.cs (hunk 2 above)
Comment (Trojaner) [reply framework/OpenMod.API/Users/UserData.cs:40]: Include ban reason if it exists too

**Comment by rube200** [comment]
i didn't test but should be all good

**Pull request merged**